src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1435,4 +1445,15 @@ public boolean isReplacementFile(DataFile df) {
         }
         return false;
     }  // end: isReplacementFile
+    
+    public List<Long> selectFilesWithMissingOriginalTypes() {
+        Query query = em.createNativeQuery("SELECT f.id FROM datafile f, datatable t where t.datafile_id = f.id AND t.originalfileformat='" + MIME_TYPE_TAB + "' ORDER BY f.id");
```

Comment:
Probably... But note that this API call is a one time fix; there will be a paragraph in the release notes for 4.7.2: "if any files have been added while your system was running 4.6.2, run this: ..., then consult the server log for any errors". And since this is something that will only need to run once, it may not be super important for it to be super neat, or super efficient, etc.